posthog/hogql_queries/trends_query_runner.py

```diff
@@ -0,0 +1,107 @@
+from typing import List, Optional, Any, Dict, Union
+
+from posthog.hogql import ast
+from posthog.hogql.parser import parse_expr, parse_select
+from posthog.hogql.query import execute_hogql_query
+from posthog.hogql.timings import HogQLTimings
+from posthog.hogql_queries.query_runner import QueryRunner
+from posthog.models import Team
+from posthog.schema import ActionsNode, EventsNode, TrendsQuery, TrendsQueryResponse
+
+
+class TrendsQueryRunner(QueryRunner):
+    query: TrendsQuery
+
+    def __init__(self, query: TrendsQuery | Dict[str, Any], team: Team, timings: Optional[HogQLTimings] = None):
+        super().__init__(team, timings)
+        if isinstance(query, TrendsQuery):
+            self.query = query
+        else:
+            self.query = TrendsQuery.parse_obj(query)
+
+    def to_query(self) -> Union[List[ast.SelectQuery], ast.SelectQuery]:
+        queries = []
+        with self.timings.measure("trends_query"):
+            for series in self.query.series:
+                queries.append(
+                    parse_select(
```

Comment:
To support multiple series. Looks like we'll have to execute multiple queries (the current trends code does this too, but in parallel with multiple series). I had looked at using a `UNION ALL` over multiple queries, but we could very easily hit max query limits given a bunch of series. Are there any alternatives that can be used here with Clickhouse / HogQL?

Reply:
Hmm... Scope creep, but what about just multiple columns in the same query? E.g. `select count(distinct person_id) as series1, count(id) as series_2, etc from event`

The base `from events` query is always the same for everyone --> all events + date range + global filters.

The individual event queries filter this further. E.g. "only $pageview with filters 1, 2, 3". This could be accomplished with something like:

```sql
select 
  countIf(distinct person_id, series_1_filters) as series1, 
  countIf(id, series_2_filters) as series_2
from event
where global_filters
```

This might break with DAU/MAU queries, but I'm not sure right now..

The global filters could then include the common filters (e.g. `event='$pageview'`) if there are any to speed up the process, otherwise they'd read through all events. Doing a UNION ALL query, or doing multiple queries and aggregating the results, works... but is likely slower as you need to do multiple passes over the data.